mappings/doab.py

```diff
@@ -1,130 +1,177 @@
-            'subjects': [('./datacite:subject/text()', '{0}||')],
-            'has_part': [(
-                './dc:identifier/text()',
-                '1|{0}|doab|text/html|{{"reader": false, "download": false, "catalog": false, "embed": true}}'
-            )],
-            'rights': [
-                (
-                    [
-                        './oaire:licenseCondition/@uri',
-                        './oaire:licenseCondition/text()'
-                    ],
-                    'doab|{0}||{1}|'
-                ),
-                (
-                    [
-                        './datacite:rights/@uri',
-                        './datacite:rights/text()'
-                    ],
-                    'doab|{0}||{1}|'
-                )
-            ]
-        }
+    def map_doab_record(self, doab_record, namespaces) -> Record:
+        identifiers, source_id = self._get_identifers(doab_record, namespaces=namespaces)
+        title = doab_record.xpath('./dc:title/text()', namespaces=namespaces) + doab_record.xpath('./datacite:creator/text()', namespaces=namespaces)
 
-    def applyFormatting(self):
-        # Set Identifiers
-        self.record.source = 'doab'
-        self.record.identifiers = self.parseIdentifiers()
+        if identifiers is None or len(identifiers) == 0 or title is None or len(title) == 0:
+            return None
 
-        # Clean up subjects
-        self.record.subjects = list(filter(lambda x: x[:3] != 'bic', self.record.subjects))
+        relations = doab_record.xpath('./dc:relation/text()', namespaces=namespaces)
+        publishers = doab_record.xpath('./dc:publisher/text()', namespaces=namespaces)
+        languages = doab_record.xpath('./dc:language/text()', namespaces=namespaces)
+        extents = doab_record.xpath('./oapen:pages/text()', namespaces=namespaces)
+        
+        return Record(
+            uuid=uuid4(),
+            frbr_status=FRBRStatus.TODO.value,
+            cluster_status=False,
+            source=Source.DOAB.value,
+            source_id=source_id,
+            identifiers= identifiers,
+            authors=self._get_authors(doab_record, namespaces=namespaces),
+            contributors=self._get_contributors(doab_record, namespaces=namespaces),
+            title=title[0],
+            is_part_of=[f'{part}||series' for part in relations if part is not None or part == ''],
+            publisher=[f'{publisher}||' for publisher in publishers if publisher is not None or publisher == ''],
+            spatial=doab_record.xpath('./oapen:placepublication/text()', namespaces=namespaces),
+            dates=self._get_dates(doab_record, namespaces=namespaces),
+            languages=[f'||{language}' for language in languages if language is not None or language == ''],
+            extent=[f'{extent} pages' for extent in extents if extent is not None or extent == ''],
+            abstract=doab_record.xpath('./dc:description/text()', namespaces=namespaces),
```

Comment:
Same as below - when we call xpath - i believe it tries to pull all elements that match that xpath. So we should just grab the first 1